enterprise/models.py

```diff
@@ -2328,3 +2331,84 @@ def __repr__(self):
         Return uniquely identifying string representation.
         """
         return self.__str__()
+
+
+class PendingEnterpriseCustomerAdminUser(TimeStampedModel):
+    """
+    Model for pending enterprise admin users.
+
+    .. pii: The user_email field contains PII, but locally deleted via
+    enterprise.signals.assign_or_delete_enterprise_admin_role when the 
+    admin registers a new account.
+    .. pii_types: email_address
+    .. pii_retirement: local_api, consumer_api
+    """
+
+    enterprise_customer = models.ForeignKey(
+        EnterpriseCustomer,
+        blank=False,
+        null=False,
+        on_delete=models.CASCADE,
+    )
+    user_email = models.EmailField(null=False, blank=False)
+    history = HistoricalRecords()
+
+    class Meta:
+        app_label = 'enterprise'
+        ordering = ['created']
+        unique_together = (('enterprise_customer', 'user_email'),)
+
+    @cached_property
+    def admin_registration_url(self):
+        """
+        Returns a URL to be used by a pending enterprise admin user to register their account.
+        """
+        registration_url = '{}/{}/admin/register'.format(
+            settings.ENTERPRISE_ADMIN_PORTAL_BASE_URL,
+            self.enterprise_customer.slug
+        )
+        return registration_url
+
+    def activate_admin_permissions(self, user, enterprise_customer):
```

Comment:
Seems like this should be a class method if we're not going to reference `self` but instead look up PendingEnterpriseCustomerAdminUsers with the given user and customer